src/textual/reactive.py

```diff
@@ -341,21 +465,76 @@ def _compute(cls, obj: Reactable) -> None:
             obj: Reactable object.
         """
         _rich_traceback_guard = True
-        for compute in obj._reactives.keys():
-            try:
-                compute_method = getattr(obj, f"compute_{compute}")
-            except AttributeError:
+        for name, reactive in obj._reactives.items():
+            if reactive._compute_method is not None:
+                compute_method = reactive._compute_method.__get__(obj)
+            else:
                 try:
-                    compute_method = getattr(obj, f"_compute_{compute}")
+                    compute_method = getattr(obj, f"compute_{name}")
                 except AttributeError:
-                    continue
+                    try:
+                        compute_method = getattr(obj, f"_compute_{name}")
+                    except AttributeError:
+                        continue
             current_value = getattr(
-                obj, f"_reactive_{compute}", getattr(obj, f"_default_{compute}", None)
+                obj, f"_reactive_{name}", getattr(obj, f"_default_{name}", None)
             )
             value = compute_method()
-            setattr(obj, f"_reactive_{compute}", value)
+            setattr(obj, f"_reactive_{name}", value)
             if value != current_value:
-                cls._check_watchers(obj, compute, current_value)
+                cls._check_watchers(obj, name, current_value)
+
+    @property
+    def watch(self) -> WatchDecorator:
+        """A decorator to make a method a watch method.
+
+        Example:
+            ```python
+            class MyWidget(Widget):
+                count = reactive(0)
+                @count.watch
+                def _count_changed(self, count:int) -> None:
+                    # Called when count changes
+                    ...
+            ```
+
```

Comment:
```suggestion
```

Reply:
Why bother? 🤷‍♂️

Reply:
I noticed it en passant and making the suggestion is almost effortless 🤷‍♂️